### PR TITLE
Add volume types filter in resource policies

### PR DESCRIPTION
**Pull request by qiuming-best**
Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)
https://github.com/vmware-tanzu/velero/issues/6482
# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: internal/resourcepolicies/volume_types_conditions.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+	if vol.HostPath != nil {
+		return "hostPath"
+	}
+	if vol.EmptyDir != nil {
+		return "emptyDir"
+	}
+	if vol.GCEPersistentDisk != nil {
+		return "gcePersistentDisk"
+	} /*
+		if vol.AWSElasticBlockStore != nil {
```
Comment (anshulahuja98) [inline internal/resourcepolicies/volume_types_conditions.go:131]: why is this commented?

**Review by anshulahuja98 — CHANGES_REQUESTED** [review]

File: internal/resourcepolicies/volume_types_conditions.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+		return "hostPath"
+	}
+	if vol.EmptyDir != nil {
+		return "emptyDir"
+	}
+	if vol.GCEPersistentDisk != nil {
+		return "gcePersistentDisk"
+	}
+	if vol.AWSElasticBlockStore != nil {
+		return "awsElasticBlockStore"
```
Comment (shubham-pampattiwar) [inline internal/resourcepolicies/volume_types_conditions.go:132]: Shouldn't we make all the PV and Volume types as constants ?

**Review by shubham-pampattiwar — COMMENTED** [review]
Added one comment, else lgtm!

File: internal/resourcepolicies/volume_types_conditions.go (hunk 2 above)
Comment (qiuming-best) [reply internal/resourcepolicies/volume_types_conditions.go]: ok, modified

File: internal/resourcepolicies/volume_types_conditions.go [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+	StorageOS            = "storageOS"
+	CSI                  = "csi"
+	EmptyDir             = "emptyDir"
+	GitRepo              = "gitRepo"
+	Secret               = "secret"
+	DownwardAPI          = "downwardAPI"
+	ConfigMap            = "configMap"
+	Projected            = "projected"
+	Ephemeral            = "ephemeral"
+)
```
Comment (mateusoliveira43) [inline internal/resourcepolicies/volume_types_conditions.go:53]: 2 suggestions:
- would be interesting to transform this in a type, like `supportedVolumes`, and change the return values of `getVolumeTypeFromPV` and `getVolumeTypeFromVolume` functions to it?
- would be interesting to sort the volumes in alphabetical order? (and also in the `getVolumeTypeFromPV` and `getVolumeTypeFromVolume` functions bodies)

Comment (qiuming-best) [reply internal/resourcepolicies/volume_types_conditions.go]: Thanks for your suggestions, I've updated

**Review by mateusoliveira43 — APPROVED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]

**Comment by draghuram** [comment]
Is this going to be merged soon? See https://kubernetes.slack.com/archives/C6VCGP4MT/p1697461184984549 for a user question in this regard.

**Review by kaovilai — APPROVED** [review]

**Review by sseago — APPROVED** [review]

**Pull request merged**